src/pools/cluster/fixed.ts

```diff
@@ -59,8 +59,8 @@ export class FixedClusterPool<
   }
 
   protected destroyWorker (worker: Worker): void {
+    worker.once('exit', () => this.removeWorker(worker))
```

Comment:
I think that callback should be added by default on both worker type. 
The issue might be that we do have a reference to the worker to remove ... 
That way, each time a worker exit for whatever reason, the circular array in the pool is freed instantly.

Reply:
Oh yay, it's working :slightly_smiling_face: